src/algebraic_topology/alternating_face_map_complex.lean

```diff
@@ -145,14 +143,39 @@ chain_complex.of_hom _ _ _ _ _ _
 
 end alternating_face_map_complex
 
-variables (C : Type*) [category C] [preadditive C]
+variables (C : Type*) [category.{v} C] [preadditive C]
 
 /-- The alternating face map complex, as a functor -/
 @[simps]
 def alternating_face_map_complex : simplicial_object C ⥤ chain_complex C ℕ :=
 { obj := alternating_face_map_complex.obj,
   map := λ X Y f, alternating_face_map_complex.map f }
 
+variables {C}
+
+def map_alternating_face_map_complex {D : Type*} [category.{v} D] [preadditive D]
```

Comment:
As you're constructing an equality, this should be a lemma, not a def. That said, I would write this as a natural isomorphism instead, as rewriting or simplifying by this equality is likely to be horrible. (If it were definitional equalities at the level of objects, you'd be okay, but it's not.)

Reply:
Thanks for the review. I use this lemma three times in my Dold-Kan equivalence code. In each case, having an equality rather than an isomorphism is certainly more convenient, and if I had to replace equalities of functors by isomorphisms, I am not even sure I would be able to make it work. The most important case is the following functoriality of the Dold-Kan equivalence, for which I do not see how I could use an isomorphism instead of an equality:
`theorem category_theory.preadditive.dold_kan.functoriality_N : ∀ {C : Type u_1} [_inst_1 : category C] [_inst_2 : additive_category C] {D : Type u_2} [_inst_3 : category D]
[_inst_4 : additive_category D] (F : C ⥤ D) [_inst_5 : F.additive],
  (simplicial_object.karoubi_whiskering C D).obj F ⋙ dold_kan.equivalence.functor =
    dold_kan.equivalence.functor ⋙ F.map_karoubi_homological_complex (complex_shape.down ℕ)`

Even if I were to relax the statement of this `functoriality_N` in order to get an isomorphism instead of an equality, the problem is that it would then be necessary to check that these isomorphisms satisfy a transitivity property with respect to the composition of additive functors `F`. With equalities of functors, this would be easy to prove using `eq_to_hom_trans`. This is a mathematical reason not to be horrified by equalities of functors.

Reply:
Okay, I'm game to try it out. :-)